### DISCUSSION TITLE
Money Flow Index (MFI)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#247] [Indicators]
Community discussion about the [Money Flow Index (MFI)](https://dotnet.stockindicators.dev/indicators/Mfi/#content)

**Comment by DaveSkender** [comment]
An interesting article on optimal parameters for MFI:
[Optimization and Testing of Money Flow Index](https://www.researchgate.net/publication/339029303_Optimization_and_Testing_of_Money_Flow_Index)

**Comment by altugavci** [comment]
I need to calculate MFI using CandePart.Close instead of CandlePart.HLC3. Are you considering adding chaining functions to this indicator?

**Reply by DaveSkender** [reply]
I'd not considered that kind of change on this one.  For indicators like MFI, I try to stay close the original author's intent -— especially when the method is backed by research or a clear design goal. Using _typical price_ is part of that foundation, and changing it could shift the indicator away from what it was designed to measure.

This might be a great idea for a [custom indicator](https://dotnet.stockindicators.dev/custom-indicators/#content).

**Reply by altugavci** [reply]
Tradingview has this function. It would be nice if you included it. I've already written a similar function. Thanks for your interest.

**Reply by DaveSkender** [reply]
Generally speaking, we’ve found that [TradingView is not a great source of truth](https://github.com/DaveSkender/Stock.Indicators/discussions/801) for correct implementation, especially for community contributed ones that have no quality control or come from less-reputable sources.  With that said, [their built-in MFI indicator](https://www.tradingview.com/support/solutions/43000502348-money-flow-mfi/) also appears to stick with using typical price.
